tests/unit/tox_recreate/plugin_test.py

```diff
@@ -0,0 +1,103 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from tox_recreate import plugin
+
+
+class TestToxConfigure:
+    def test_when_theres_no_cached_hash_file(self, config):
+        plugin.tox_configure(config)
+
+        assert config.envconfigs["test"].recreate is True
+
+    def test_when_the_hash_doesnt_match(self, config, cached_hash_path):
+        cached_hash_path.write_text("wrong_hash")
+
+        plugin.tox_configure(config)
+
+        assert config.envconfigs["test"].recreate is True
+
+    def test_when_the_hash_matches(self, config, cached_hash_path, setup_cfg_hash):
+        cached_hash_path.write_text(setup_cfg_hash)
+
+        plugin.tox_configure(config)
+
+        assert not config.envconfigs["test"].recreate
+
+    @pytest.fixture
+    def config(self, make_envconfig):
+        """Return a mock tox Config object."""
+        return MagicMock(
+            spec_set=["envconfigs", "envlist"],
+            envconfigs={
+                "format": make_envconfig("format"),
+                "test": make_envconfig("test"),
+            },
+            envlist=["test"],
+        )
```

Comment:
I looked into adding `tox` itself as a test dependency so that the tests could import it and autospec these tox objects but it doesn't gain you anything: `create_autospec()` returns a mock that doesn't have `envconfigs` or `envlist` attrs. It was the same with the other mock tox objects below. Lord knows how these classes are implemented inside tox